src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3576,7 +3576,9 @@ void ride_construction_toolupdate_construct(int32_t screenX, int32_t screenY)
         return;
     }
 
-    for (;;)
+    // search for appropriate z value for ghost, up to max ride height
+    int numAttempts = (z <= (8 * MAX_TRACK_HEIGHT) ? MAX_TRACK_HEIGHT - (z / 8) + 1 : 2);
```

Comment:
If you are going to make this change make the change for mazes as well its just inconsistent as it is at present.
As this is a complex equation it should really be a constexpr that is named.